### PR TITLE
adapter: rename arrangement merge effort option

### PR DESCRIPTION
**Pull request by teskje**
This PR renames the cluster/replica option `IDLE ARRANGEMENT MERGE EFFORT` to just `ARRANGEMENT MERGE EFFORT`. Dropping the "idle" removes a potential source of confusion as the configured merge effort isn't actually applied only when a worker is idle, but on every worker step.

Furthermore, and perhaps more importantly, this PR changes the `ARRANGEMENT MERGE EFFORT` cluster/replica option to control the new `arrangement_exert_proportionality` instead of the deprecated `idle_arrangement_merge_effort`. The plan is to remove `idle_arrangement_merge_effort` in a follow-up.

We also stop documenting the arrangement merge effort cluster/replica option here. It's an internal option and we don't expect users to care about it, much less understand what it does.

### Motivation

  * This PR adds a known-desirable feature.

Part of #17002

### Tips for reviewer

I suggest looking at the commits separately.

The changes made here are pretty mechanical, they just make for a large diff because of all the renaming in adapter and the catalog migration. Nothing should be surprising here, I hope!

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Review by umanwizard — APPROVED** [review]

**Comment by teskje** [comment]
There is the open question of whether we actually need a replica option for the proportionality. Maybe having the feature flag is enough? Moving that back into draft until that's answered.

File: src/stash/src/upgrade/v43_to_v44.rs [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+fn convert_replica_config(config: &v43::ReplicaConfig) -> v44::ReplicaConfig {
+    use v43::replica_config::Location as Location42;
+    use v44::replica_config::Location as Location43;
+
+    let arrangement_exert_proportionality =
+        translate_idle_merge_effort(config.idle_arrangement_merge_effort.as_ref());
+    let location = match &config.location {
+        Some(Location42::Managed(v)) => Some(Location43::Managed(WireCompatible::convert(v))),
+        Some(Location42::Unmanaged(v)) => Some(Location43::Unmanaged(WireCompatible::convert(v))),
```
Comment (def-) [inline src/stash/src/upgrade/v43_to_v44.rs:130]: Uncovered

File: src/stash/src/upgrade/v43_to_v44.rs [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        arrangement_exert_proportionality,
+        location,
+    }
+}
+
+fn translate_idle_merge_effort(effort: Option<&v43::ReplicaMergeEffort>) -> Option<u32> {
+    match effort.as_ref().map(|e| e.effort) {
+        Some(0) => Some(0),
+        Some(value) => {
+            warn!(%value, "untranslatable `idle_arrangement_merge_effort`");
```
Comment (def-) [inline src/stash/src/upgrade/v43_to_v44.rs:145]: uncovered

File: src/sql/src/plan/statement/ddl.rs [hunk 3]
```diff
@@ -4263,9 +4263,9 @@ pub fn plan_alter_cluster(
             if let Some(availability_zones) = availability_zones {
                 options.availability_zones = AlterOptionParameter::Set(availability_zones);
             }
-            if let Some(idle_arrangement_merge_effort) = idle_arrangement_merge_effort {
-                options.idle_arrangement_merge_effort =
-                    AlterOptionParameter::Set(idle_arrangement_merge_effort);
+            if let Some(arrangement_exert_proportionality) = arrangement_exert_proportionality {
+                options.arrangement_exert_proportionality =
```
Comment (def-) [inline src/sql/src/plan/statement/ddl.rs:4267]: uncovered

File: src/sql/src/plan/statement/ddl.rs [hunk 4]
```diff
@@ -4224,9 +4224,9 @@ pub fn plan_alter_cluster(
                     if replication_factor.is_some() {
                         sql_bail!("REPLICATION FACTOR not supported for unmanaged clusters");
                     }
-                    if idle_arrangement_merge_effort.is_some() {
+                    if arrangement_exert_proportionality.is_some() {
                         sql_bail!(
-                            "IDLE ARRANGEMENT MERGE EFFORT not supported for unmanaged clusters"
+                            "ARRANGEMENT EXERT PROPORTIONALITY not supported for unmanaged clusters"
```
Comment (def-) [inline src/sql/src/plan/statement/ddl.rs:4229]: uncovered

**Review by def- — APPROVED** [review]
Nightly looks fine (same as on main): https://buildkite.com/materialize/nightlies/builds/5140
Coverage had a weird error I haven't seen before: https://buildkite.com/materialize/coverage/builds/312#_
```
 thread 'test_persist_debug' panicked at 'snapshot assertion for 'persist_opened_trace' failed in line 149', /cargo/registry/src/index.crates.io-6f17d22bba15001f/insta-1.33.0/src/runtime.rs:563:9 
```
I have retriggered it, maybe just an unrelated flake. Coverage results are marked inline, but only some minor edge cases it seems.

**Comment by teskje** [comment]
Closing in favor of https://github.com/MaterializeInc/materialize/pull/26319.

**Pull request closed without merging**